src/figuresClasses.ts

```diff
@@ -1,19 +1,88 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('length cant be 0 ');
+    }
 
-export class Circle {
+    if (this.a + this.b <= this.c
+      || this.c + this.a <= this.b
+      || this.b + this.c <= this.a) {
+      throw new Error(
+        'the longest side of a triangle is >= than a sum of two other',
+      );
+    }
+  }
 
+  getArea(): number {
+    const semiperimeter: number = (this.a + this.b + this.c) / 2;
+    const area: number = (
+      Math.sqrt(
+        semiperimeter
+        * (semiperimeter - this.a)
+        * (semiperimeter - this.b)
+        * (semiperimeter - this.c),
+      )
+    );
+
+    return Math.floor(area * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?